pkg/ccl/backupccl/restore_job.go

```diff
@@ -1631,6 +1635,30 @@ func (r *restoreResumer) doResume(ctx context.Context, execCtx interface{}) erro
 	return nil
 }
 
+func clusterRestoreDuringUpgradeErr(
+	clusterVersion roachpb.Version, binaryVersion roachpb.Version,
+) error {
+	return errors.Errorf("cluster restore not supported during major version upgrade: restore started at cluster version %s but binary version is %s",
+		clusterVersion,
+		binaryVersion)
+}
+
+// validateJobDetails returns an error if this job cannot be resumed.
+func (r *restoreResumer) validateJobIsResumable(execConfig *sql.ExecutorConfig) error {
```

Comment:
nit: might be cleaner to have the following helper func that can get called in planning and in execution:

`checkMixedVersionClusterRestore(clusterVersion roachpb.Version, binaryVersion roachpb.Version)`